### ISSUE TITLE
Not everything is aware of where it's running from

### ISSUE THREAD

**Issue by brianjgeiger** [brianjgeiger/osf.io#28]
If we're going to have the ability to run via the OSF and via a Django App, we'll need to ensure that everything works in both places.  Currently, the Links class is returning port 8000 for all links regardless of if it's running on 5000 or 8000. Except for the HTML links, which are going to 5000. The Links should match the port it's running on. Similarly, the swagger documentation is not prepending the /api/v2 route when running as an OSF application but works fine when running on 8000 which does not have that route.

The first part of this is to document all of the inconsistencies. The second is to fix the inconsistencies. The third is to have a way (documentation, methodology, better code, or whatever) to prevent future features from having these problems.

**Comment by brianjgeiger** [comment]
Steve says, "Former issue probably has to do with this setting: https://github.com/CenterForOpenScience/osf.io/blob/4a18effa8162646ba5dc228c8bb7a673882cad76/api/base/settings.py#L103 , which allows the browseable API to work on /api/v2/ . When we make the API public facing, we'll probably remove that setting (/api/v2/) is for internal use only." (via https://github.com/brianjgeiger/osf.io/issues/27#issuecomment-99488275 )

**Comment by brianjgeiger** [comment]
Is this something you could look into @samanehsan? At least the first two parts.

**Comment by samanehsan** [comment]
Sure, I'll look into it after #19 if that's okay.

**Comment by brianjgeiger** [comment]
If you think you'll lose a lot of momentum switching, then go for it, but this one is much more of a blocker than #19 is.

**Comment by samanehsan** [comment]
Ok, in that case I can look into this first.

**Comment by samanehsan** [comment]
Since the swagger documentation still doesn't work when running the API as an OSF application, I'll make a separate issue for it.

**Comment by brianjgeiger** [comment]
That's great, thank you.
